utils/experiment/experiment.go

```diff
@@ -45,6 +45,9 @@ func IsTerminating(experiment *v1alpha1.Experiment) bool {
 		case v1alpha1.AnalysisPhaseFailed, v1alpha1.AnalysisPhaseError, v1alpha1.AnalysisPhaseInconclusive:
 			return true
 		}
+		if run.Phase == v1alpha1.AnalysisPhaseSuccessful && run.RequiredForCompletion {
+			return true
+		}
```

Comment:
This doesn't seem correct. This logic states that if **_any one_** of the N number of analyses runs which required for completion is successful, then we are saying that the entire experiment is about to finish. It should only do this if all of them are completed and successful.

Reply:
fixed